tests/CMakeLists.txt

```diff
@@ -192,6 +194,7 @@ foreach(test ${tests})
         link_directories(${test} PRIVATE "${CMAKE_SOURCE_DIR}/../lib")
     endif()
   endif()
+  set_target_properties(${test} PROPERTIES COMPILE_DEFINITIONS "${libzmq_DEFS}" )
```

Comment:
I don't think this is the adequate way to propagate compile definitions to dependent targets.

target_compiler_definitions(libzmq PUBLIC ...) should be used somewhere above instead, then they should be propagated automatically.

See https://cmake.org/cmake/help/v3.0/command/target_compile_definitions.html

Reply:
Thanks, I'll have a look at `PUBLIC`.

Reply:
Nice! `PUBLIC` is a very handy feature.